src/main/java/org/hisp/dhis/rules/parser/expression/function/OperatorCompareEqual.java

```diff
@@ -0,0 +1,54 @@
+package org.hisp.dhis.rules.parser.expression.function;
+
+/*
+ * Copyright (c) 2004-2020, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import org.hisp.dhis.antlr.AntlrExpressionVisitor;
+import org.hisp.dhis.antlr.operator.AntlrOperatorCompareEqual;
+import org.hisp.dhis.parser.expression.antlr.ExpressionParser;
+import org.hisp.dhis.rules.parser.expression.CommonExpressionVisitor;
+
+/**
+ * Compare operator: equal
+ *
+ * @author Jim Grace
```

Comment:
I suggest you change the author in all the classes like this to Zubair Asghar. Even though you are just writing a wrapper around a class that I wrote, it is still your wrapper class. :)

It seems a shame to need to create all these classes as wrappers for classes that already exist in dhis2-antrl-expression-parser. But I can't think of a better way. (I tried!) We can't use the same technique that we do in dhis2-core of an interface with a default method (see interface ExpressionItem), because this is not supported in all versions of Java where the Android app may have to run.

Perhaps at some time we can refactor the classes in dhis2-antlr-expression-parser to have a getDescription() method as well, since this is needed both for dhis2-core and now for dhis2-rule-engine. But if you don't want to significantly extend the scope of this feature, this could be done as a refactor at a later time. I think what you have done is a fine solution for now.

Reply:
Having getDescription in `dhis2-antlr-expression-parser` would an ideal solution.